src/guide/lists.md

```diff
@@ -23,6 +23,18 @@ let oneTwoThree = [1, ...twoThree]
 print(oneTwoThree) // [1, 2, 3]
 ```
 
+For convenience, it is also possible to use the spread syntax at any position in a list:
+
+```grain
+let oneTwo = [1, 2]
+let threeFour = [3, 4]
+let result = [...oneTwo, ...threeFour, 5]
+
+print(result) // [1, 2, 3, 4, 5]
+```
+
+However, it is important to be aware of the **performance implications of arbitrary-position spreads**: in general, including a single spread at the end of a list expression is much more efficient than including a spread anywhere else as the latter requires copying the entire list(s) being spread.
```

Comment:
This makes a lot of sense to someone who's familiar with these kinds of lists in fp languages, but I think this is still going to be a bit of a head-scratcher for someone coming from, say, JavaScript. We probably need to make this its own section and explain a bit more of why this is the case.